pkg/stanza/fileconsumer/reader.go

```diff
@@ -84,24 +94,88 @@ func (r *Reader) ReadToEnd(ctx context.Context) {
 			break
 		}
 
+		var prevHeaderFinalized = r.HeaderFinalized
```

Comment:
Rather than track two separate things, can we just have one flag on the reader that indicates that it's time to update the scanner, and clear the flag while doing so?

Reply:
Great idea, done!